File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/order/adaptor/OrderAdaptor.java
package band.gosrock.domain.domains.order.adaptor;


import band.gosrock.common.annotation.Adaptor;
import band.gosrock.domain.domains.order.domain.Order;
import band.gosrock.domain.domains.order.exception.OrderNotFoundException;
import band.gosrock.domain.domains.order.repository.OrderRepository;
import band.gosrock.domain.domains.order.repository.condition.FindEventOrdersCondition;
import band.gosrock.domain.domains.order.repository.condition.FindMyPageOrderCondition;
import java.util.List;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;

@Adaptor
@RequiredArgsConstructor
public class OrderAdaptor {

    private final OrderRepository orderRepository;

    public Order save(Order order) {
        return orderRepository.save(order);
    }

    public Order findById(Long orderId) {
        return orderRepository
                .findById(orderId)
                .orElseThrow(() -> OrderNotFoundException.EXCEPTION);
    }

    public List<Order> findByEventId(Long eventId) {
        return orderRepository.findByEventId(eventId);
    }

    public Order findByOrderUuid(String uuid) {
        return orderRepository
                .findByOrderUuid(uuid)
                .orElseThrow(() -> OrderNotFoundException.EXCEPTION);
    }

    public List<Order> findByUuidIn(List<String> orderUuids) {
        return orderRepository.findByUuidIn(orderUuids);
    }

    public Optional<Order> findRecentOrderByUserId(Long userId) {
        return orderRepository.findRecentOrder(userId);
    }

    public Slice<Order> findMyOrders(FindMyPageOrderCondition condition, Pageable pageable) {
        return orderRepository.findMyOrders(condition, pageable);
    }

    public Page<Order> findEventOrders(FindEventOrdersCondition condition, Pageable pageable) {
        return orderRepository.findEventOrders(condition, pageable);
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/order/adaptor/OrderAdaptor.kt
package band.gosrock.domain.domains.order.adaptor

import band.gosrock.common.annotation.Adaptor
import band.gosrock.domain.domains.order.domain.Order
import band.gosrock.domain.domains.order.domain.OrderStatus
import band.gosrock.domain.domains.order.domain.RefundStatus
import band.gosrock.domain.domains.order.exception.OrderNotFoundException
import band.gosrock.domain.domains.order.repository.OrderRepository
import band.gosrock.domain.domains.order.repository.condition.FindEventOrdersCondition
import band.gosrock.domain.domains.order.repository.condition.FindMyPageOrderCondition
import java.util.Optional
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice

@Adaptor
class OrderAdaptor(private val orderRepository: OrderRepository) {

    fun save(order: Order): Order = orderRepository.save(order)

    fun findById(orderId: Long): Order =
        orderRepository.findById(orderId).orElseThrow { OrderNotFoundException.EXCEPTION }

    fun findByEventId(eventId: Long): List<Order> =
        orderRepository.findByEventId(eventId)

    fun findByOrderUuid(uuid: String): Order =
        orderRepository.findByOrderUuid(uuid).orElseThrow { OrderNotFoundException.EXCEPTION }

    fun findByUuidIn(orderUuids: List<String>): List<Order> =
        orderRepository.findByUuidIn(orderUuids)

    fun findRecentOrderByUserId(userId: Long): Optional<Order> =
        orderRepository.findRecentOrder(userId)

    fun findMyOrders(condition: FindMyPageOrderCondition, pageable: Pageable): Slice<Order> =
        orderRepository.findMyOrders(condition, pageable)

    fun findEventOrders(condition: FindEventOrdersCondition, pageable: Pageable): Page<Order> =
        orderRepository.findEventOrders(condition, pageable)

    fun findByEventIdAndOrderStatus(eventId: Long, orderStatus: OrderStatus): List<Order> =
        orderRepository.findByEventIdAndOrderStatus(eventId, orderStatus)

    fun findByEventIdAndOrderStatusAndUserId(eventId: Long, userId: Long, orderStatus: OrderStatus): List<Order> =
        orderRepository.findByEventIdAndUserIdAndOrderStatus(eventId, userId, orderStatus)

    fun findRefunds(eventId: Long?, refundStatus: RefundStatus?, keyword: String?, pageable: Pageable): Page<Order> =
        orderRepository.findRefunds(eventId, refundStatus, keyword, pageable)
}
